exporter/fileexporter/file_exporter.go

```diff
@@ -101,3 +135,19 @@ func (e *fileExporter) Start(context.Context, component.Host) error {
 func (e *fileExporter) Shutdown(context.Context) error {
 	return e.file.Close()
 }
+
+func buildMarshaler(marshalType string) (ptrace.Marshaler, pmetric.Marshaler, plog.Marshaler) {
+	if strings.ToLower(marshalType) == marshalTypeProto ||
+		strings.ToLower(marshalType) == marshalTypeProtobuf {
+		return ptrace.NewProtoMarshaler(), pmetric.NewProtoMarshaler(), plog.NewProtoMarshaler()
+	}
+	return ptrace.NewJSONMarshaler(), pmetric.NewJSONMarshaler(), plog.NewJSONMarshaler()
```

Comment:
We should add some validation of the parameter to `Config.Validate`, so that the user cannot submit bad inputs.